lib/browser/api/auto-updater/squirrel-update-win.ts

```diff
@@ -15,89 +15,67 @@ const isSameArgs = (args: string[]) => args.length === spawnedArgs.length && arg
+        detached,
         windowsHide: true
       });
       spawnedArgs = args || [];
     }
-  } catch (error1) {
-    error = error1 as Error;
-
-    // Shouldn't happen, but still guard it.
-    process.nextTick(function () {
-      return callback(error);
-    });
-    return;
-  }
-  stdout = '';
-  stderr = '';
 
-  spawnedProcess.stdout.on('data', (data) => { stdout += data; });
-  spawnedProcess.stderr.on('data', (data) => { stderr += data; });
+    let stdout = '';
+    let stderr = '';
 
-  errorEmitted = false;
-  spawnedProcess.on('error', (error) => {
-    errorEmitted = true;
-    callback(error);
-  });
-
-  return spawnedProcess.on('exit', function (code, signal) {
-    spawnedProcess = undefined;
-    spawnedArgs = [];
-
-    // We may have already emitted an error.
-    if (errorEmitted) {
-      return;
-    }
+    spawnedProcess.stdout.on('data', (data) => { stdout += data; });
+    spawnedProcess.stderr.on('data', (data) => { stderr += data; });
 
-    // Process terminated with error.
-    if (code !== 0) {
-      return callback(`Command failed: ${signal ?? code}\n${stderr}`);
-    }
+    spawnedProcess.on('error', (error) => {
+      reject(error);
+    });
 
-    // Success.
-    callback(null, stdout);
+    spawnedProcess.on('exit', function (code, signal) {
+      spawnedProcess = undefined;
+      spawnedArgs = [];
+
+      if (code !== 0) {
+        // Process terminated with error.
+        reject(new Error(`Command failed: ${signal ?? code}\n${stderr}`));
+      } else {
+        // Success.
+        resolve(stdout);
+      }
+    });
   });
 };
 
 // Start an instance of the installed app.
 export function processStart () {
-  return spawnUpdate(['--processStartAndWait', exeName], true, function () {});
+  spawnUpdate(['--processStartAndWait', exeName], true);
 }
 
 // Download the releases specified by the URL and write new results to stdout.
-export function checkForUpdate (updateURL: string, callback: (error: Error | null, update?: any) => void) {
-  return spawnUpdate(['--checkForUpdate', updateURL], false, function (error: Error, stdout: string) {
-    let ref, ref1, update;
-    if (error != null) {
-      return callback(error);
-    }
-    try {
-      // Last line of output is the JSON details about the releases
-      const json = stdout.trim().split('\n').pop();
-      update = (ref = JSON.parse(json!)) != null ? (ref1 = ref.releasesToApply) != null ? typeof ref1.pop === 'function' ? ref1.pop() : undefined : undefined : undefined;
-    } catch {
-      return callback(new Error(`Invalid result:\n${stdout}`));
-    }
-    return callback(null, update);
-  });
+export async function checkForUpdate (updateURL: string): Promise<any> {
+  const stdout = await spawnUpdate(['--checkForUpdate', updateURL], false);
+  try {
+    // Last line of output is the JSON details about the releases
+    const json = stdout.trim().split('\n').pop();
+    JSON.parse(json!)?.releasesToApply?.pop?.();
```

Comment:
Should we validate that `json` is non-null?

Reply:
I didn't want to change the logic here, `JSON.parse()` will throw and we are catching the error, so it's not necessary. We could throw a custom error, but we always turn in into `Invalid result:\n${stdout}` anyway